signature/src/policy.rs

```diff
@@ -0,0 +1,825 @@
+                // One accepted signature is enough.
+                Ok(()) => {
+                    return Ok(());
+                }
+                Err(e) => {
+                    reject_reason.push(e);
+                }
+            }
+        }
+
+        if reject_reason.is_empty() {
+            reject_reason.push(anyhow!("Can not find any signatures!"));
+        }
+
+        Err(anyhow!(format!(
+            "The signatures do not satisfied! Reject reason: {:?}",
+            reject_reason
+        )))
+    }
+
+    #[allow(unused_assignments)]
+    fn is_signature_accept(&self, image: &image::Image, sig: Vec<u8>) -> Result<()> {
+        // FIXME: only support "GPGKeys" type now.
+        if self.key_type != *KEY_TYPE_GPG {
+            return Err(anyhow!(
+                "UNKOWN KEY TYPE in policy config: only support 'GPGKeys' now!"
+            ));
+        }
+
+        if !self.key_path.is_empty() && !self.key_data.is_empty() {
+            return Err(anyhow!(
+                "Internal inconsistency: both \"keyPath\" and \"keyData\" specified!"
+            ));
+        }
+
+        let mut pubkey_ring = Vec::new();
+        if !self.key_data.is_empty() {
+            // ???: base64 encode the keyring data ???
+            // FIXME: Maybe the KeyData format need to re-think.
+            pubkey_ring = base64::decode(&self.key_data)?;
+        } else {
+            pubkey_ring = fs::read(&self.key_path)?;
+        }
+
+        // Verify the signature with the pubkey ring.
+        let sig_payload = signatures::verify_sig_and_extract_payload(pubkey_ring, sig)?;
+
+        // Verify whether the information recorded in signature payload
+        // is consistent with the real information of the image.
+        //
+        // The match policy of image-reference is the "signedIdentity" field.
+        sig_payload
+            .validate_signed_docker_reference(&image.reference, self.signed_identity.clone())?;
+        sig_payload.validate_signed_docker_manifest_digest(&image.manifest_digest)?;
+
+        Ok(())
+    }
+}
+
+// PolicyReferenceMatch specifies a set of image identities(image-reference) accepted in PolicyRequirement.
+#[derive(Serialize, Deserialize, Debug, PartialEq, Clone)]
+#[serde(untagged)]
+pub enum PolicyReferenceMatch {
+    // PrmRemapIdentity: "remapIdentity".
+    RemapIdentity(PrmRemapIdentity),
+    // PrmExactRepository: "exactRepository".
+    ExactRepository(PrmExactRepository),
+    // PrmExactReference: "exactReference".
+    ExactReference(PrmExactReference),
+    // PrmMatch: "matchExact", "matchRepoDigestOrExact", "matchRepository".
+    Match(PrmMatch),
+}
+
+impl Default for PolicyReferenceMatch {
+    fn default() -> Self {
+        PolicyReferenceMatch::Match(PrmMatch {
+            type_name: PRM_TYPE_MATCH_REPO_DIGEST_OR_EXACT.to_string(),
+        })
+    }
+}
+
+impl PolicyReferenceMatch {
+    pub fn matchs_docker_reference(
```

Comment:
Sorry as this is pedantic, but `matches_docker_reference` might be a better name here, in PrmMatch, PrmExactReference, PrmExactRepository and PrmRemapIdentity